src/watermarks.ml

```diff
@@ -246,10 +329,17 @@ let subst ?name vcs =
            (fun () -> Vcs.commit_id vcs))
       (fun () -> Vcs.files vcs)
   in
-  let name = get_name ~files ?name () in
+  let dune_project =
+    if List.mem ~set:files Dune_project.file then
```

Comment:
I didn't notice this before, but it would be better if `~files` was a set or we passed the comparison function explicitly.

Reply:
I changed this to `List.exists`. Making it a set doesn't seem super useful given that this is the only membership test we perform